tests/components/sensor/test_islamic_prayer_times.py

```diff
@@ -0,0 +1,160 @@
+"""The tests for the Islamic prayer times sensor platform."""
+from datetime import datetime, timedelta
+from unittest.mock import patch
+from homeassistant.setup import async_setup_component
+from homeassistant.components.sensor.islamic_prayer_times import \
+    IslamicPrayerTimesData
+from tests.common import MockDependency
+import homeassistant.util.dt as dt_util
+from tests.common import async_fire_time_changed
+
+LATITUDE = 41
+LONGITUDE = -87
+CALC_METHOD = 'isna'
+PRAYER_TIMES = {"Fajr": "06:10", "Sunrise": "07:25", "Dhuhr": "12:30",
+                "Asr": "15:32", "Maghrib": "17:35", "Isha": "18:53",
+                "Midnight": "00:45"}
+
+
+def get_prayer_time_as_dt(prayer_time):
+    """Create a datetime object for the respective prayer time."""
+    today = datetime.today().strftime('%Y-%m-%d')
+    date_time_str = '{} {}'.format(str(today), prayer_time)
+    pt_dt = dt_util.parse_datetime(date_time_str)
+    return pt_dt
+
+
+async def test_islamic_prayer_times_min_config(hass):
+    """Test minimum Islamic prayer times configuration."""
+    config = {
+        'sensor': {
+            'platform': 'islamic_prayer_times'
+        }
+    }
+    assert await async_setup_component(hass, 'sensor', config) is True
+
+
+async def test_islamic_prayer_times_multiple_sensors(hass):
+    """Test Islamic prayer times sensor with multiple sensors setup."""
+    config = {
+        'sensor': {
+            'platform': 'islamic_prayer_times',
+            'sensors': [
+                'fajr', 'sunrise', 'dhuhr', 'asr', 'maghrib', 'isha',
+                'midnight'
+            ]
+        }
+    }
+    assert await async_setup_component(hass, 'sensor', config) is True
+
+
+async def test_islamic_prayer_times_with_calculation_method(hass):
+    """Test Islamic prayer times configuration with calculation method."""
+    config = {
+        'sensor': {
+            'platform': 'islamic_prayer_times',
+            'calculation_method': 'mwl'
+        }
+    }
+    assert await async_setup_component(hass, 'sensor', config) is True
+
+
+async def test_islamic_prayer_times_data_init():
```

Comment:
Remove this test and instead test that the correct state is reported for the calculation method in the test above.